tests/e2e/e2e_util_test.go

```diff
@@ -1044,3 +1103,134 @@ func (s *IntegrationTestSuite) execWithdrawReward(
 	s.executeGaiaTxCommand(ctx, c, gaiaCommand, valIdx, endpoint)
 	s.T().Logf("Successfully withdrew distribution rewards for delegator %s from validator %s", delegatorAddress, validatorAddress)
 }
+
+// register a ica on chainB from registrant on chainA
+func (s *IntegrationTestSuite) submitICAtx(owner, connectionID, txJsonPath string) {
+	fee := sdk.NewCoin(uatomDenom, math.NewInt(930000))
+	s.T().Logf("register an interchain account on chain %s for %s from chain %s", s.chainB.id, owner, s.chainA.id)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	submitTX := []string{
+		"gaiad",
+		"tx",
+		"icamauth",
+		"submit",
+		txJsonPath,
+		fmt.Sprintf("--%s=%s", flags.FlagFrom, owner),
+		fmt.Sprintf("--%s=%s", "connection-id", connectionID),
+		fmt.Sprintf("--%s=%s", flags.FlagGas, flags.GasFlagAuto),
+		fmt.Sprintf("--%s=%s", flags.FlagFees, fee),
+		fmt.Sprintf("--%s=%s", flags.FlagChainID, s.chainA.id),
+		"--keyring-backend=test",
+		"--broadcast-mode=sync",
+		"--output=json",
+		"-y",
+	}
+
+	exec, err := s.dkrPool.Client.CreateExec(docker.CreateExecOptions{
+		Context:      ctx,
+		AttachStdout: true,
+		AttachStderr: true,
+		Container:    s.valResources[s.chainA.id][0].Container.ID,
+		User:         "nonroot",
+		Cmd:          submitTX,
+	})
+	s.Require().NoError(err)
+
+	var (
+		outBuf bytes.Buffer
+		errBuf bytes.Buffer
+	)
+
+	err = s.dkrPool.Client.StartExec(exec.ID, docker.StartExecOptions{
+		Context:      ctx,
+		Detach:       false,
+		OutputStream: &outBuf,
+		ErrorStream:  &errBuf,
+	})
+
+	s.Require().NoErrorf(
+		err,
+		"failed to submit ica tx; stdout: %s, stderr: %s", outBuf.String(), errBuf.String(),
```

Comment:
I think it's better if we can avoid manipulating a Read/Write stream and save the result into a string or byte variable. We use the stream into a closure. This can cause race conditions.

Reply:
what do you mean by race condition ?  

outBuf is `bytes.Buffer` which is a slice of bytes.

Reply:
I think this is addressed in the e2e refactor